tests/installation/system_role.pm

```diff
@@ -44,6 +45,7 @@ sub assert_system_role {
 }
 
 sub run {
+    # Define default role
```

Comment:
for me this is more readable version of this module , doing exactly the same as this 2 function version....
```
sub run {
    # Define default role
    # Still initializing the system at this point, can take some time
    # Verify default role for SLE15, it's text for sles and gnome for sled
    assert_screen 'system-role-default-system', 180;    
    if (get_var('SYSTEM_ROLE')) {
        send_key 'alt-' . $role_hotkey{get_var('SYSTEM_ROLE')};
        assert_screen "system-role-$system_role-selected";
        # every system role other than default will end up in textmode
        set_var('DESKTOP', 'textmode');
    }
    send_key $cmd{next};
}
```

Reply:
That's not exactly same, in your code we select role even if it's 'default', which is equivalent to empty, so we have to check both.